mm2src/coins/z_coin/z_rpc.rs

```diff
@@ -295,8 +323,17 @@ impl SaplingSyncLoopHandle {
         let request = tonic::Request::new(ChainSpec {});
         let current_blockchain_block = self.grpc_client.get_latest_block(request).await?;
         let current_block_in_db = block_in_place(|| self.blocks_db.get_latest_block())?;
+        let extrema = block_in_place(|| self.wallet_db.lock().block_height_extrema())?;
+
+        let mut from_block = self
+            .consensus_params
+            .sapling_activation_height
+            .max(current_block_in_db as u32 + 1) as u64;
```

Comment:
I think this can lead to `attempt to add with overflow`, if `current_block_in_db: i64 > u32::MAX`.